tests/utils.py

```diff
@@ -87,6 +92,169 @@ def make_nifti_image(array, affine=None):
     return image_name
 
 
+class DistTestCase(unittest.TestCase):
+    """testcase without _outcome, so that it's picklable."""
+
+    original_mp = None
+
+    def setUp(self) -> None:
+        self.original_mp = torch.multiprocessing.get_start_method(allow_none=True)
+        try:
+            torch.multiprocessing.set_start_method("spawn", force=True)
+        except RuntimeError:
+            pass
+
+    def tearDown(self) -> None:
+        try:
+            torch.multiprocessing.set_start_method(str(self.original_mp), force=True)
```

Comment:
Hi @ericspod ,

This is not the recommended `launch method` in PyTorch distributed data parallel, do you think it's OK to test with it? I am not very sure about it. Others looks good to me.

Thanks.

Reply:
Hi @Nic-Ma, the main difference is that `torch.distributed.launch` is a simple wrapper of
```py
process = subprocess.Popen(cmd, env=current_env, stdout=stdout_handle, stderr=stderr_handle)
```
this PR's test util is doing
```py
p = torch.multiprocessing.Process(target=self.run_process, args=(func, proc_rank, args, kwargs, results))
```
I agree that `torch.distributed.launch` is closer to the real-world use cases,
but the test util is easier to use for wrapping the test cases, testing out the basic gloo/nccl backend usages